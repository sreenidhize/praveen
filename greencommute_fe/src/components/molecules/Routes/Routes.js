import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from "@material-ui/core";
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import FiberManualRecordIcon from '@material-ui/icons/FiberManualRecord';
import RoomOutlinedIcon from '@material-ui/icons/RoomOutlined';
import CloseOutlinedIcon from '@material-ui/icons/CloseOutlined';

const COLOR_Iceberg = '#e3f3f6';

const useStyles = makeStyles((theme)=>({
    container: {
        maxWidth: "330px",
        height: "50px",
        margin: "15px 0px",
        position: "relative",
        borderRadius: "10px",
        background: COLOR_Iceberg,
    },
    destination: {
        maxWidth: "330px",
        height: "50px",
        position: "relative",
        borderRadius: "10px",
        background: COLOR_Iceberg,
    },
    location: {
        display: "flex",
        transform: "translateY(-50%)",
        position: "absolute",
        top: "50%",
    },
    icon: {
        marginTop: "-3px",
        marginLeft: "8px",
        color: theme.typography.subtitle1.color,
    },
    text: {
        marginLeft: "10px",
    },
    closeIcon: {
        position: "relative",
        right:"-20px",
    }

}));

const Routes = (props) => {
    const classes = useStyles();
    const { from, to }=props;
    return (
        <Grid container direction="column" data-testid="Routes-placeholder">
            <img src="https://i.stack.imgur.com/8kOXQ.jpg" alt="Commute Map" width="330px" height="135px"></img>
            <Grid container className={classes.container} >
                <Grid item className={classes.location}>
                    <FiberManualRecordIcon className={classes.icon} />
                    <Typography variant="body1" className={classes.text}>{from}</Typography>
                    <CloseOutlinedIcon className={`${classes.icon} ${classes.closeIcon}`}/>
                </Grid>
            </Grid>
            <Grid container className={classes.destination}>
                <Grid item className={classes.location}>
                    <RoomOutlinedIcon className={classes.icon} />
                    <Typography variant="body1" className={classes.text}>{to}</Typography>
                </Grid>
            </Grid>
        </Grid >

    );
}

Routes.propTypes = {
    from: PropTypes.string,
    to: PropTypes.string,
}

export default Routes;