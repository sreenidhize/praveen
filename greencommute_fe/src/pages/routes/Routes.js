import { Typography } from "@material-ui/core";
import React from "react";
import { Route, Switch } from "react-router-dom";
import LandingPage from "../LandingPage/LandingPage";
import JobSearch from "../../components/templates/jobs-search/JobSearch";

function Routes() {
  return (
    <Switch>
      <Route exact path="/jobsearch">
        {/* render jobSearch page */}
        <JobSearch shouldShowSearch={true} />
      </Route>
      <Route exact path="/savedjobs">
        {/* render job serach with only saved jobs */}
        <JobSearch shouldShowSearch={false} />
      </Route>
      <Route exact path="/">
        <LandingPage />
      </Route>
      <Route path="*">
        <Typography children="Page Not Found." />
      </Route>
    </Switch>
  );
}

export default Routes;
