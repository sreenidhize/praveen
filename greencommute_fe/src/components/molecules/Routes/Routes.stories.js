import React from 'react';
import Routes from './Routes';

export default {
    title: 'Molecules/Routes',
    component: Routes
}

const Template = (args) =><Routes {...args} />;

export const Primary = Template.bind({});
Primary.args = {
    from: "East Marredpally, E Marredpally",
    to: "Hitech City, Telangana, Secunderabad",
};
