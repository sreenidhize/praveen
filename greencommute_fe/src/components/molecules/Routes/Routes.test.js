import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Routes from './Routes'

describe("Routes component test", () => {
  it("Matches the snapshot",() => {
    const { container } = render(
      <Routes />
    );
    expect(container).toMatchSnapshot();
  });

  test("test Routes render",() => {
    const { getByTestId } = render( <Routes />);
    expect(getByTestId("Routes-placeholder")).toBeInTheDocument();
  });
});