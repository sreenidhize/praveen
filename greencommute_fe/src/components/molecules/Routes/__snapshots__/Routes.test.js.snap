// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes component test Matches the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
    data-testid="Routes-placeholder"
  >
    <img
      alt="Commute Map"
      height="135px"
      src="https://i.stack.imgur.com/8kOXQ.jpg"
      width="330px"
    />
    <div
      class="MuiGrid-root makeStyles-container-1 MuiGrid-container"
    >
      <div
        class="MuiGrid-root makeStyles-location-3 MuiGrid-item"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-icon-4"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <circle
            cx="12"
            cy="12"
            r="8"
          />
        </svg>
        <p
          class="MuiTypography-root makeStyles-text-5 MuiTypography-body1"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-icon-4 makeStyles-closeIcon-6"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
          />
        </svg>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-destination-2 MuiGrid-container"
    >
      <div
        class="MuiGrid-root makeStyles-location-3 MuiGrid-item"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-icon-4"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zM7 9c0-2.76 2.24-5 5-5s5 2.24 5 5c0 2.88-2.88 7.19-5 9.88C9.92 16.21 7 11.85 7 9z"
          />
          <circle
            cx="12"
            cy="9"
            r="2.5"
          />
        </svg>
        <p
          class="MuiTypography-root makeStyles-text-5 MuiTypography-body1"
        />
      </div>
    </div>
  </div>
</div>
`;
